editors/neovim/files/patch-lua51-includes.diff

```diff
@@ -0,0 +1,14 @@
+diff --git CMakeLists.txt CMakeLists.txt
+index 12e0d6e6a..cbbe716dd 100644
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -8,6 +8,8 @@
+ # Version should match the tested CMAKE_URL in .githuworkflows/build.yml.
+ cmake_minimum_required(VERSION 3.13)
+
++include_directories(BEFORE @PREFIX@/include/lua5.1)
```

Comment:
If neovim uses `luajit` as its runtime, should `/opt/local/include/luajit-2.1` be used here instead?

Reply:
No, the issue is with the [nlua0](https://github.com/neovim/neovim/blob/8df6736ca14d09f87cf0a8486758ac5708819434/src/nlua0.c) C library that neovim builds, and then subsequently uses with luajit. `nlua0` is getting built against the lua 5.3 headers from the `lua` port, and then luajit chokes on `nlua0` because it is compiled against the wrong version of lua.

(I documented all of my findings extensively in [#70420](https://trac.macports.org/ticket/70420).